tools/proto_format/proto_sync.py

```diff
@@ -12,7 +12,8 @@
 
 
 def git_status(path):
-    return subprocess.check_output(['git', 'status', '--porcelain', str(path)]).decode()
+    return subprocess.check_output(
+        ['git', 'status', '--porcelain', str(path)], cwd=str(path)).decode()
```

Comment:
should we change the workdir by the `cwd`? Looks like the `git_status` is used to check if there is dirty api tree.

Reply:
im not sure i follow - but the thing that this script is passed is the api_root and that is what it is comparing - so i used this 

the other option would be to use the api_root parent which is the repo - but this felt a bit arbitrary so just used the api_root instead

Reply:
`str(path)` is absolute so it will check the (correct) api path - but do it _in_ that directory also

Reply:
its the equivalent of the following in bash

```console

x@y:/src/workspace/envoy# cd api
x@y:/src/workspace/envoy/api# git status --porcelain /src/workspace/envoy/api

```

Reply:
Yep, it's OK if the `str(path)` is absolute. But in the current code base, the default value of `api_root` is `./api`, a normal relatively string may be passed to the script.

So I am worried that this change will result in some unexpected results.

```
    parser.add_argument('--api_root', default='./api')
```

Reply:
ok lets remove the default and i guess make the arg compulsory

Reply:
hmm, actually wait - rather just make sure the path is absolute - it is always called so in ci, but for sure we should convert whatever we are passed in to being absolute - good catch

Reply:
Thanks. 😄  

FWIW, how about change the
                 
```
_git_status = git_status(api_root)
```
to
```
_git_status = git_status(api_root_path)
```
where the `api_root_path` always a absolute version.

Reply:
oh wait again!

that already happens - see here https://github.com/envoyproxy/envoy/blob/main/tools/proto_format/proto_sync.py#L47

maybe that should happen in main to be more explicit - i can cleanup when i follow up - im working on these files atm

Reply:
k - got it - on it - thanks for thorough review